src/Qir/Runtime/lib/QIR/QubitManager.cpp

```diff
@@ -241,14 +240,14 @@ namespace Quantum
                                                                 sharedQubitStatusArray);
     }
 
-    Qubit CQubitManager::Allocate()
+    QubitIdType CQubitManager::Allocate()
     {
         QubitIdType newQubitId = AllocateQubitId();
         FailIf(newQubitId == NoneMarker, "Not enough qubits.");
-        return CreateQubitObject(newQubitId);
+        return newQubitId;
     }
 
-    void CQubitManager::Allocate(Qubit* qubitsToAllocate, int32_t qubitCountToAllocate)
+    void CQubitManager::Allocate(QubitIdType* qubitsToAllocate, QubitIdType qubitCountToAllocate)
```

Comment:
qubitCountToAllocate should still be int32_t.

Reply:
In theory changing this back to `int32_t` would mean a caller could never allocate all the qubits at once, since max qubit id is larger than max value for `int32_t`. In practice this will likely never be a problem, but it's worth noting that this is another of places where there is a mismatch between the max the type can support and the max that the code can actually accommodate. The types below or similar, where changing them back to smaller integers would imply a restriction. If I can get it compile that way I can make the shift and put it up for PR so we can discuss, but I figured it was worth calling out this implied restriction.